src/brevitas/nn/quant_sdpa.py

```diff
@@ -0,0 +1,209 @@
+            attn_mask (optional Tensor): Attention mask; shape must be broadcastable to the shape of attention weights,
+                which is :math:`(N,..., L, S)`. Two types of masks are supported.
+                A boolean mask where a value of True indicates that the element *should* take part in attention.
+                A float mask of the same type as query, key, value that is added to the attention score.
+            dropout_p (float): Dropout probability; if greater than 0.0, dropout is applied
+            is_causal (bool): If set to true, the attention masking is a lower triangular matrix when the mask is a
+                square matrix. The attention masking has the form of the upper left causal bias due to the alignment
+                (see :class:`torch.nn.attention.bias.CausalBias`) when the mask is a non-square matrix.
+                An error is thrown if both attn_mask and is_causal are set.
+            scale (optional float, keyword-only): Scaling factor applied prior to softmax. If None, the default value is set
+                to :math:`\frac{1}{\sqrt{E}}`.
+            enable_gqa (bool): Ignored to make calling interface compatible with PyTorch >v2.5. Always set to False.
+
+        Returns:
+            output (Tensor): Attention output; shape :math:`(N, ..., Hq, L, Ev)`.
+
+        Shape legend:
+            - :math:`N: \text{Batch size} ... : \text{Any number of other batch dimensions (optional)}`
+            - :math:`S: \text{Source sequence length}`
+            - :math:`L: \text{Target sequence length}`
+            - :math:`E: \text{Embedding dimension of the query and key}`
+            - :math:`Ev: \text{Embedding dimension of the value}`
+            - :math:`Hq: \text{Number of heads of query}`
+            - :math:`H: \text{Number of heads of key and value}`
+        """
+        kwargs = {}
+        if scale is not None:
+            kwargs["scale"] = scale
+        if enable_gqa:
+            kwargs["enable_gqa"] = enable_gqa
+        return F.scaled_dot_product_attention(
+            query=query,
+            key=key,
+            value=value,
+            attn_mask=attn_mask,
+            dropout_p=dropout_p,
+            is_causal=is_causal,
+            **kwargs)
+
+
+class QuantScaledDotProductAttention(Module):
+
+    def __init__(
+            self,
+            softmax_input_quant=None,
+            attn_output_weights_quant=Uint8ActPerTensorFloat,
+            q_scaled_quant=Int8ActPerTensorFloat,
+            k_transposed_quant=Int8ActPerTensorFloat,
+            v_quant=Int8ActPerTensorFloat,
+            attn_output_quant=None,
+            **kwargs) -> None:
+        super(QuantScaledDotProductAttention, self).__init__()
+
+        def filter_kwargs(prefix):
+            return {k[len(prefix):]: v for k, v in kwargs.items() if k.startswith(prefix)}
+
+        self.q_scaled_quant = QuantIdentity(act_quant=q_scaled_quant, **filter_kwargs('q_scaled_'))
```

Comment:
This should work, but I believe we also have a QuantLayer argument to specify the prefix, which should be equivalent. Rarely used so it's worth double checking.

Reply:
Added test for this.